cmd/convert.go

```diff
@@ -53,6 +55,52 @@ var convertCmd = &cobra.Command{
 	Short: "Convert a Docker Compose file",
 	PreRun: func(cmd *cobra.Command, args []string) {
 
+		// if the profile flag is provided
+		if GlobalProfile != "" {
+			viper.SetConfigType("yaml")
+			viper.SetConfigName("kompose")
+			viper.AddConfigPath(".")
+			viper.AddConfigPath("$HOME/.kompose/")
+
+			customPrefFile := os.Getenv("KOMPOSE_CONFIG")
+			if customPrefFile != "" {
+				viper.SetConfigFile(customPrefFile)
+			}
+
+			if err := viper.ReadInConfig(); err != nil {
+				logrus.Fatalf("Could not read config file: %v", err)
+			}
+
+			profile := "profiles." + GlobalProfile
+			// profile not found
+			if viper.Get(profile) == nil {
+				logrus.Fatalf("Given profile %q not found in any of config files.", GlobalProfile)
+			}
+
+			// find out what provider information is given in profile
+			provider := profile + ".provider"
+			viper.BindPFlag(provider, cmd.Flags().Lookup("provider"))
+			GlobalProvider = viper.GetString(provider)
+
+			// find what controllers are mentioned
+			controllers := profile + ".objects"
+
+			for _, controller := range viper.GetStringSlice(controllers) {
+				switch controller {
+				case "deployment":
+					ConvertDeployment = true
+				case "replicationcontroller":
+					ConvertReplicationController = true
+				case "deploymentconfig":
+					ConvertDeploymentConfig = true
+				case "daemonset":
+					ConvertDaemonSet = true
+				default:
+					logrus.Fatalf("Invalid controller %q mentioned in profile.", controller)
+				}
+			}
+		}
+
```

Comment:
All this code above should be refactored into a separate function. Feels messy here.